ReactAndroid/src/main/java/com/facebook/react/modules/network/OkHttpClientProvider.java

```diff
@@ -90,12 +89,8 @@ public static OkHttpClient createClient() {
 
         client.sslSocketFactory(new TLSSocketFactory(trustManager), trustManager);
 
-        ConnectionSpec cs = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
-                .tlsVersions(TlsVersion.TLS_1_0, TlsVersion.TLS_1_1, TlsVersion.TLS_1_2)
-                .build();
-
         List<ConnectionSpec> specs = new ArrayList<>();
-        specs.add(cs);
+        specs.add(ConnectionSpec.MODERN_TLS);
         specs.add(ConnectionSpec.COMPATIBLE_TLS);
         specs.add(ConnectionSpec.CLEARTEXT);
```

Comment:
Now you have same as [`OkHttpClient.DEFAULT_CONNECTION_SPECS`](https://github.com/square/okhttp/blob/okhttp_36/okhttp/src/main/java/okhttp3/OkHttpClient.java#L125-L126).

    static final List<ConnectionSpec> DEFAULT_CONNECTION_SPECS = Util.immutableList(
      ConnectionSpec.MODERN_TLS, ConnectionSpec.COMPATIBLE_TLS, ConnectionSpec.CLEARTEXT);

Reply:
I'll test this with default specs and we can remove this if it works for all devices. Important part is the TlsSocketFactory, which enables TLS1_1 & TLS1_2 on older devices. Before this PR, we were disabling TLS1_1 and TLS1_0 for newer ones, but it's better to stick with default specs from okhttp

Reply:
I removed the custom connection specs overall, and tested on KitKat, this seems to be working as expected.